Controller/Annotations/AbstractScalarParam.php

```diff
@@ -51,6 +51,12 @@ public function getConstraints()
                 'pattern' => '#^(?:'.$this->requirements['rule'].')$#xsu',
                 'message' => $this->requirements['error_message'],
             ));
+        } elseif (is_array($this->requirements)) {
+            foreach ($this->requirements as $requirement) {
+                if ($requirement instanceof Constraint) {
```

Comment:
IMO we should deprecate having something else than an array of Constraints at this point.

Reply:
How would you like to see that implemented? trigger a E_USER_DEPRECATED?

Reply:
Yes :)

Reply:
Done

Reply:
Travis fails because of the new deprecation notices. [Symfony recommends](https://symfony.com/doc/current/components/phpunit_bridge.html#trigger-deprecation-notices) supressing the notices using the `@` prefix. Shall I do that? Or is there a way for PHPUnit to expect these deprecations?

Reply:
Please silence them with `@` indeed

Reply:
Done

Reply:
@sandermarechal but travis CI is still failing, isn't it?